event-sourcing/event-store-api/src/main/scala/org/apache/james/eventsourcing/eventstore/EventStore.scala

```diff
@@ -36,4 +38,7 @@ trait EventStore {
   def appendAll(events: Iterable[Event]): Publisher[Void]
 
   def getEventsOfAggregate(aggregateId: AggregateId): Publisher[History]
+
+  def removeBefore(before: Instant): Publisher[Void]
```

Comment:
Should be

```suggestion
  def remove(aggregateId: AggregateId): Publisher[Void]
```

Reply:
I'm not clear your mean, we don't need `beforeDate` ?

Reply:
No. We will list old tasks, then we will destroy there additional informations in here based on AggregateId.

No need date.